dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/CondJumpModel.java

```diff
@@ -0,0 +1,37 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.Register.Read;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.IfAsJump;
+
+import java.util.Set;
+import java.util.List;
+
+
+public class CondJumpModel extends DefaultEventModel implements RegReaderModel {
+    public CondJumpModel(CondJump event) {
+        super(event);
+    }
+
+    public boolean isGoto() {
+        return ((CondJump) event).isGoto();
+    }
+
+    public boolean isDead() {
+        return ((CondJump) event).isDead();
+    }
+
+    public List<Event> getDependentEvents() {
+        if (event instanceof IfAsJump jump) {
+            return jump.getBranchesEvents();
+        }
+        return ((CondJump) event).getSuccessor().getSuccessors();
+    }
```

Comment:
I think it is awkward that this method returns `Events` rather than `EventModels`. That means, you will in general obtain events that are outside of the execution. I think this should be avoided.

Generally, you should think about what do you actually need to capture about the dynamic(!) execution of an event.

Reply:
It is easy to get `Event` from `EventModel` but one have to query `ExecutionModelNext` to get the `EventModel` from an `Event`, i.e. we can let it return `EventModels` but `ExecutionModelNext` should be a parameter of this method, is it OK to do so? I think it is awkward also...